semantic/fb_types.go

```diff
@@ -0,0 +1,16 @@
+package semantic
+
+import (
+	flatbuffers "github.com/google/flatbuffers/go"
+	"github.com/influxdata/flux/semantic/internal/fbsemantic"
+)
+
+type fbTabler interface {
+	Init(buf []byte, i flatbuffers.UOffsetT)
+	Table() flatbuffers.Table
+}
+
+type FBMonoType struct {
+	mt  fbsemantic.MonoType
+	tbl fbTabler
+}
```

Comment:
Would you be opposed to calling this `MonoType`? Also why do we need both an `fbTabler` and an `fbsemantic.MonoType`? Isn't `fbsemantic.MonoType` an `fbTabler`?

Reply:
In the Go generated code, `MonoType` is just a kind integer that describes which specific monotype something is:
https://github.com/influxdata/flux/blob/master/semantic/internal/fbsemantic/MonoType.go#L5-L14

In fact there is no interface generated by `flatc` that unites `Arr`, `Fun`, etc as the same kind of thing.  `fbTabler` defines any FlatBuffers type that is defined as a `table`.

Yes, I see that `MonoType` is not already taken in the semantic package, so I can rename it to `MonoType`.

Reply:
I could have just omitted the `fbsemantic.MonoType` here, since we can use Go's runtime type casting to get at `Arr`, `Fun`, `Var`, etc.  But I thought it might be useful, maybe.